miniwin/miniwin/src/miniwin_d3drmviewport.cpp

```diff
@@ -29,16 +29,6 @@ Direct3DRMViewportImpl::~Direct3DRMViewportImpl()
 	FreeDeviceResources();
 }
 
-void Direct3DRMViewportImpl::CollectSceneData(IDirect3DRMFrame* group)
-{
-	m_backgroundColor = static_cast<Direct3DRMFrameImpl*>(group)->m_backgroundColor;
-
-	std::vector<PositionColorVertex> vertices =
-		{{-1, -1, 0, 0, 255, 0, 255}, {1, -1, 0, 0, 0, 255, 255}, {0, 1, 0, 255, 0, 0, 128}};
-
-	PushVertices(vertices.data(), vertices.size());
-}
-
```

Comment:
```suggestion
void Direct3DRMViewportImpl::CollectSceneData(IDirect3DRMFrame* group)
{
	m_backgroundColor = static_cast<Direct3DRMFrameImpl*>(group)->m_backgroundColor;

	std::vector<PositionColorVertex> vertices =
		{{-1, -1, 0, 0, 255, 0, 255}, {1, -1, 0, 0, 0, 255, 255}, {0, 1, 0, 255, 0, 0, 128}};

	PushVertices(vertices.data(), vertices.size());
}

```

Reply:
I guess now that my PR has been merged you can refactor it in a way you think makes sens since it's a bit more clear what I was aiming for.

Reply:
My goal here was to create a geometry buffer for every frame, such that `PushVertices/CollectSceneData` would be necessary anymore.
But I ran into an issue about where to store this data. The api makes it possible to have multiple devices, to which you can all render a frame. So I'd have to move the buffer to the device. This can probably be implemented through smart data structures, but I did not think it through yet.